qiskit_nature/problems/sampling/protein_folding/peptide/beads/base_bead.py

```diff
@@ -0,0 +1,89 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""An abstract class defining a bead of a peptide."""
+from abc import ABC
+from typing import List, Tuple, Union, Callable
+
+from qiskit.opflow import PauliOp, OperatorBase
+
+from ..pauli_ops_builder import _build_full_identity
+from ...residue_validator import _validate_residue_symbol
+
+
+class BaseBead(ABC):
+    """An abstract class defining a bead of a peptide."""
+
+    def __init__(
+        self,
+        chain_type: str,
+        main_index: int,
+        residue_type: str,
+        turn_qubits: List[PauliOp],
+        build_turn_indicator_fun_0: Callable,
+        build_turn_indicator_fun_1: Callable,
+        build_turn_indicator_fun_2: Callable,
+        build_turn_indicator_fun_3: Callable,
+    ):
+        """
+        Args:
+            chain_type: Type of the chain, either "main_chain" or "side_chain".
+            main_index: index of the bead on the main chain in a peptide.
+            residue_type: A character representing the type of a residue for the bead.
+            turn_qubits: A list of Pauli operators that encodes the turn following from a
+                            given bead index.
+            build_turn_indicator_fun_0: method that build turn indicator functions for the bead.
+            build_turn_indicator_fun_1: method that build turn indicator functions for the bead.
+            build_turn_indicator_fun_2: method that build turn indicator functions for the bead.
+            build_turn_indicator_fun_3: method that build turn indicator functions for the bead.
```

Comment:
Could we provide a little more detail what these functions do/look like? Maybe providing an example would be helpful, even if it simply links to another method (or the docstring thereof).